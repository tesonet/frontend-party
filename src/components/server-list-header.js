import React from "react";
import {connect} from "react-redux";
import styled from "styled-components";

import logoImg from "../assets/logo-testio-dark.png";
import logoutIcon from "../assets/logout-icon.svg";
import logoutIconHover from "../assets/logout-icon-hover.svg";
import {setLogged} from "../redux/actions";


const Header = styled.div`
    width: 100%;
    height: 113px;
    border-bottom: 1px solid #e6e6e6;
`;

const Wrapper = styled.div`
    display: flex;
    margin-bottom: 70px;
    padding: 0 15px;
    justify-content: space-between;
    align-items: center;
    height: 100%;
`;

const Logout = styled.button`
    cursor: pointer;
    color: #2f3254;
    font-weight: 300;
    background: white;
    border: none;
    font-size: 16px;
    &:before {
        width: 16px;
        content: url(${logoutIcon});
        position: relative;
        left: -10px;
        top: 2px;
        height: 20px;
    }
    &:hover {
        color: #99cc33;
    }
    &:focus {
        outline: none;
    }
    &:hover:before {
        content: url(${logoutIconHover});
    }
`;

export const ServerListHeader = (props) => {

    const logout = () => {
        props.setLogged(false);
        localStorage.removeItem('token');
    };

    return (
        <Header>
            <Wrapper>
                <img src={logoImg} alt="testio logo"/>
                <Logout onClick={logout}>Logout</Logout>
            </Wrapper>
        </Header>
    );
};

export default connect(
    null,
    {setLogged}
)(ServerListHeader);
