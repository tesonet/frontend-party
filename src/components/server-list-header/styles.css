.container {
  width: 100vw;
  height: 114px;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.logo {
  height: 30px;
  width: 130px;
  margin: 42px 0 42px 15px;
  flex: 1;
}

.logoutContainer {
  display: flex;
  justify-content: flex-end;
  width: 100%;
  flex: 10;
}

.button {
  width: 90px;
  height: 30px;
  margin-right: 20px;
  padding: 0 10px 0 30px;
  color: #2f3254;
  background: url("../../../assets/logout.svg") no-repeat scroll 10px;
  background-color: Transparent;
  outline:none;
  border: none;
  cursor: pointer;
  font-size: 14px;
  font-weight: 300; 
  letter-spacing: 0.35px;
  line-height: 30px;
}

.passwordInput {
  background-color: #ffff;
}

.button:hover {
  background-color: #99cc33;
}