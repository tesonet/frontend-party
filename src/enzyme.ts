import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetchMock from "jest-fetch-mock";

declare let global: { fetch: {} };

global.fetch = fetchMock;
configure({ adapter: new Adapter() });
