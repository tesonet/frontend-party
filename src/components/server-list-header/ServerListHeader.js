import React, { useState } from "react";
import PropTypes from "prop-types";
import { Redirect } from "react-router-dom";
import testioLogo from "../../../assets/testio-dark.png";

import styles from "./styles.css";

const ServerListHeader = ({ logoutAction }) => {
  const [logout, setLogout] = useState(false);
  return logout ? (
    <Redirect to="/" />
  ) : (
    <div className={styles.container}>
      <img className={styles.logo} src={testioLogo} alt="testio logo" />
      <div className={styles.logoutContainer}>
        <button
          type="button"
          className={styles.button}
          onClick={() => {
            logoutAction();
            setLogout(true);
          }}
        >
          Logout
        </button>
      </div>
    </div>
  );
};

ServerListHeader.propTypes = {
  logoutAction: PropTypes.func.isRequired
};

export default ServerListHeader;
