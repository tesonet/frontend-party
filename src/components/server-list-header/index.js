import { connect } from "react-redux";
import { logoutAction } from "../../_actions/auth";

import ServerListHeader from "./ServerListHeader";

const mapDispatchToProps = {
  logoutAction
};

export default connect(
  null,
  mapDispatchToProps
)(ServerListHeader);
