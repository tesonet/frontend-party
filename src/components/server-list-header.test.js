import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";

import {ServerListHeader} from "./server-list-header";


const component = shallow(<ServerListHeader/>);

it('renders correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
});
