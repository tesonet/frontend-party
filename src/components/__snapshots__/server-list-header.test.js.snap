// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="testio logo"
      src=""
    />
    <styled.button
      onClick={[Function]}
    >
      Logout
    </styled.button>
  </styled.div>
</styled.div>
`;
